File: mv_tool.py
from pathlib import Path
import shutil
from check_file_permission import check_file_permission

def mv_file(src_path, dst_path, config):
    if not check_file_permission(src_path, "write", config) or not check_file_permission(dst_path, "write", config):
        raise PermissionError("Access denied for source or destination")
    src_path, dst_path = Path(src_path).resolve(), Path(dst_path).resolve()
    if not src_path.is_file():
        raise ValueError(f"Source file not found: {src_path}")
    shutil.move(src_path, dst_path)
    return f"Moved {src_path} to {dst_path}"

File: ts-tools/mv_tool.ts

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function mvFile(srcPath: string, dstPath: string, config: any): Promise<string> {
    return new Promise((resolve, reject) => {
        Promise.all([
            checkFilePermission(srcPath, "write", config),
            checkFilePermission(dstPath, "write", config)
        ])
            .then(() => {
                const resolvedSrc = path.resolve(srcPath);
                const resolvedDst = path.resolve(dstPath);
                fs.rename(resolvedSrc, resolvedDst, (err) => {
                    if (err) {
                        reject(err);
                    } else {
                        resolve(`Moved ${srcPath} to ${dstPath}`);
                    }
                });
            })
            .catch(reject);
    });
}
